Makefile

```diff
@@ -0,0 +1,91 @@
+
+VERSION := 0.1.0
+
+# Repository
+SRC_DIR := src
+
+PARSER_REPO_URL   ?= $(shell git -C $(SRC_DIR) remote get-url origin)
+$(eval PARSER_NAME=$(shell basename $(PARSER_REPO_URL) | cut -d '-' -f3 | sed 's#.git##' ))
+UPPER_PARSER_NAME := $(shell echo $(PARSER_NAME) | tr a-z A-Z)
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect C++ sources, and link if necessary
+CPPSRC := $(wildcard $(SRC_DIR)/*.cc)
+
+ifeq (, $(CPPSRC))
+	ADDITIONALLIBS := 
+else
+	ADDITIONALLIBS := -lc++
+endif
+
+# collect sources
+SRC := $(wildcard $(SRC_DIR)/*.c)
+SRC += $(CPPSRC)
+OBJ := $(addsuffix .o,$(basename $(SRC)))
+
+# ABI versioning
```

Comment:
When would I bump these?

Reply:
(Referring to the ABI versions, if that wasn't clear)

Reply:
not sure myself. just leave it with 0 is just fine I'd say.